yarn-project/archiver/src/archiver/kv_archiver_store/block_store.ts

```diff
@@ -235,6 +239,14 @@ export class BlockStore {
     void this.#lastProvenL2Block.set(blockNumber);
   }
 
+  getProvenL2EpochNumber(): number {
+    return this.#lastProvenL2Epoch.get() ?? 0;
+  }
```

Comment:
Seems a little scary to return 0 instead of undefined here. How would we distinguish between a proven epoch 0 and an unproven epoch 0?

```suggestion
  getProvenL2EpochNumber(): number | undefined {
    return this.#lastProvenL2Epoch.get();
  }
```

Reply:
Yes, good spot.